PortalFetch/crawler_sel.py

```diff
@@ -0,0 +1,332 @@
+    driver.find_element_by_id("j_password").send_keys(password)
+    driver.find_element_by_id(
+        "btn-eventId-proceed").click()  # Wait for the response from the next page and make sure the page is loaded!
+    logging.info("Log in finished.")
+
+
+# Open course search page
+def openSearchPage(driver):
+    """
+    Open search page.
+
+    Click Appls->Look Up Classes-> open search page
+    Parameters: webdriver
+    Returns: None
+    """
+    # Find the Apps menu and click
+    findAppsMenu(driver)
+    # Find in App List（Look Up Classes）button
+    classes = lookUpClasses(driver)
+    mainWindowName = driver.window_handles[0]  # Main window name
+
+    classes.click()
+    time.sleep(2)  # Make sure the new window is open
+    #
+    # Switch to the newly opened window, because this window is the course search page, you must switch the program pointer to this window to operate this window
+    windowNames = driver.window_handles  # Get all window names
+    for name in windowNames:
+        if mainWindowName == name:
+            continue  # Do not switch to the main window because the search page opens from the main window
+
+        driver.switch_to_window(name)
+        break  # Because there are only two windows: the main window and the open window, switching directly to the end is over
+    # Waiting for elements in the page to appear, indicating that the page has finished loading
+    waitUtilPageLoaded(driver)
+
+
+def findAppsMenu(driver):
+    """
+    Find Apps menu.
+
+    Parameters: webdriver
+    Returns: None
+    """
+    menus = driver.find_elements_by_class_name("list-group-item")
+    if not menus:
+        logging.info("No left menu found！")
+        return
+    appMenu = []
+    for menu in menus:
+        txt = menu.text
+        if ("apps".lower() == txt.lower()):  # No left menu found
+            appMenu = menu
+            break
+    if not appMenu:
+        logging.info("Apps menu not found in left menu！")
+        return
+    time.sleep(2)  # Wait for the next page to come over
+    appMenu.click()  # Open the menu
+
+
+def lookUpClasses(driver):
+    """
+    Find app list.
+
+    Parameters: webdriver
+    Returns: classes
+    Return type: list
+    """
+    clss = driver.find_elements_by_class_name("myapps-item")
+    t = type(clss)
+    print(t)
+    if not clss:
+        logging.info("No app list found！")
+        return
+    classes = []
+    for cls in clss:
+        txt = cls.find_element_by_class_name("myapps-item-label").text
+        txt = str.strip(txt)
+        if ("Look Up Classes".lower() == txt.lower()):
+            classes = cls
+            break
+    if not classes:
+        logging.info("No Look Up Classes feature found in the app list!")
+        return
+    return classes
+
+
+# Go to the advanced options page and start filling in various search terms
+def fillAdvanceSearch(driver):
+    """
+    Fill advance search.
+
+    Parameters: webdriver
+    Returns: None
+    """
```

Comment:
```
"""Fill advance search.

Go to the advanced options page and start filling in various search terms. [More details..]

Args:
    driver: ....

Returns: 
    None
"""
```
Refer to https://github.com/google/styleguide/blob/gh-pages/pyguide.md  `3.8.3 Functions and Methods`.

Similar things for other functions.

Reply:
Thanks! But it looks that put the first line to the second line is fine for our test.
"""
Fill advance search.

Go to the advanced options page and start filling in various search terms. [More details..]

Args:
    driver: ....

Returns: 
    None
"""